assets/dev/js/editor/controls/base.js

```diff
@@ -126,6 +126,13 @@ ControlBaseView = Marionette.CompositeView.extend( {
 
 		var isVisible = elementor.helpers.isActiveControl( this.model, settings.attributes, settings.controls );
 
+		const parentContainerType = this.model.get( 'parent_container_type' );
+
+		if ( isVisible && parentContainerType ) {
+			isVisible = this.container.settings.get( 'container_type' ) === parentContainerType ||
```

Comment:
In which scenario will we need to check the item itself?
AFAIU we need to check only the parent item

Reply:
@davseve The container itself is also a "widget". So we need to update the flex based controls in the container widget itself's Advanced tab also.